tools/istio-iptables/pkg/capture/run_test.go

```diff
@@ -337,6 +337,43 @@ func TestSeparateV4V6(t *testing.T) {
 	}
 }
 
+func TestCleanup(t *testing.T) {
+	cases := []struct {
+		name   string
+		config func(cfg *config.Config)
+	}{
+		{
+			"cleanup-empty",
+			func(cfg *config.Config) {
+				cfg.PreemptiveCleanup = true
+			},
+		},
+		{
+			"cleanup-dns",
```

Comment:
Given my comments elsewhere, could we add another test (here or elsewhere) that validates that IF an `ISTIO_xxx` chain + jump exists, the jump and chain are removed even if the rules inside the chain are not exactly what we expect?

Reply:
I will add a test case for this.
Right now I'm struggling with the existing .golden files that needs to be updated to due the cleanup being enabled by default.
Do you know if the .golden files for tools/istio-iptables are supposed to be generated via `make refresh-goldens` ? It looks like the content has to be manually updated  :|

Reply:
> Do you know if the .golden files for tools/istio-iptables are supposed to be generated via `make refresh-goldens` ? It looks like the content has to be manually updated :|

For iptables they are manually created, yes. Which is good, since it helps catch problems where expectation does not match reality :D

Reply:
> I guess that, considering the way the reverse logic operates, we are always placed in a sort-of-scenario where "rules inside the chain are not exactly what we expect" simply because the "reverser" ignores them and continues to remove the custom chains regardless of the rules inside them.

Yep, I just want to add a test that would explicitly fail if someone in the future breaks that behavior, since it's a bit subtle.